snapcraft/internal/repo/_deb.py

```diff
@@ -186,20 +188,91 @@ def get_packages_for_source_type(cls, source_type):
         return packages
 
     @classmethod
-    def install_build_packages(cls, package_names):
-        unique_packages = set(package_names)
+    def _setup_multi_arch(cls, apt_cache, package_name):
```

Comment:
Given the complexity (not the code, but the task, it is invasive after all), can we add a docstring?